### DISCUSSION TITLE
LM Studio Granite GGUF model loading error

### DISCUSSION THREAD

**Discussion by dukeinps** [ibm-granite/discussions#7] [Q&A] [answered]
Thank you for the GGUF models on huggingface. They download and have the correct parameters to run but but sadly they fail to load in LM Studio. Getting this error:
"llama.cpp error: 'check_tensor_dims: tensor 'output.weight' not found'" In researching this on the llama.cpp github discussions, there is no clear answer. The message infers that something is missing or incomplete.

**Comment by dprosper** [comment]
similar issue attempting to use Ollama, the error is sliglty different then above. 
```
error loading model vocabulary: unknown pre-tokenizer type: 'refact
```
in contrast one from instructlab [granite-7b-lab-GGUF](https://huggingface.co/instructlab/granite-7b-lab-GGUF) works fine.

**Comment by mayank31398** [comment, marked as answer]
Hi, currently you need to use llama.cpp from main branch I think.
I am unsure if ollama is supported currently.

**Comment by akiemiq4g** [comment]
I am running on Ollama pre-release version 0.1.39 and that is how I got around @dprosper the refact error but the chat response has not been properly parsed. Therefore, it looks like this but I bet it probably looks better with the proprietary app IBM developed to use their model: 
<img width="1703" alt="Screenshot 2024-05-28 at 3 50 20 PM" src="https://github.com/ibm-granite/discussions/assets/12855567/02a3cfb3-e80a-48b2-b29a-428456139b10">

I feel like I'll end up just downloading the IBM app just to get my feet wet with it but not a fan of this initial road of installing it. But if you intend on integrating it into an app and will be doing parsing downstream of your api calls then this should work for you. I just had to download this version: https://github.com/ollama/ollama/releases/tag/v0.1.39 and install it by dragging it into my Applications folder on Mac overwriting my previous 0.1.38 version. I will probably revert back as I'd rather run on the stable version as I'm not developing apps that require any pre-release functionality as of yet and not sure if it has caused any side-effects to the other models I've been using via Ollama server.

**Reply by mayank31398** [reply]
I see. I have tried LM Studio btw.
Its working fine for me.
